pkg/config/config.go

```diff
@@ -117,82 +126,50 @@ func GetYamlFromFile(cfgPath string, k0sVars constant.CfgVars) (clusterConfig *v
 
 func ValidateYaml(cfgPath string, k0sVars constant.CfgVars) (clusterConfig *v1beta1.ClusterConfig, err error) {
 	var storage *v1beta1.StorageSpec
+	var cfg *v1beta1.ClusterConfig
+
+	CfgFile = cfgPath
+
 	if k0sVars.DefaultStorageType == "kine" {
 		storage = &v1beta1.StorageSpec{
 			Type: v1beta1.KineStorageType,
 			Kine: v1beta1.DefaultKineConfig(k0sVars.DataDir),
 		}
-
 	}
-	switch cfgPath {
-	case "-":
-		clusterConfig, err = v1beta1.ConfigFromStdin(storage)
-	case "":
-		clusterConfig = v1beta1.DefaultClusterConfig(storage)
-	default:
-		clusterConfig, err = v1beta1.ConfigFromFile(cfgPath, storage)
+	if file.Exists(constant.K0sConfigPathDefault) {
+		logrus.Debugf("found config file in %s", constant.K0sConfigPathDefault)
 	}
+	cfgReader, err := getConfigReader()
```

Comment:
the `cfgReader` may be an open file, but is never closed?

We may need move the logic of `getConfigReader` to here, and add a `defer cfgReader.Close()`. Note that we should not close `stdin`.